### PR TITLE
Feature - Weather icons

### PR DESCRIPTION
**Pull request by KlaudiaLum**

### CODE REVIEW

File: src/app/weather-icon/weather-icon.component.ts [hunk 1]
```diff
@@ -28,15 +26,16 @@ export class WeatherIconComponent implements OnInit {
...
 
-  private generateRandomCoordinates(): void {
-
-    this.randomLatitude = Math.random() * (90 - (-90)) + (-90);
-    this.randomLongitude = Math.random() * (180 - (-180)) + (-180);
+  generateRandomCoordinates(): void {
+    // Generating random geographical coordinates
+    this.randomLatitude = Math.random() * 180 - 90;
+    this.randomLongitude = Math.random() * 360 - 180;
+
```
Comment (GustavBlomqvist) [inline src/app/weather-icon/weather-icon.component.ts]: Remove this

File: src/app/weather-icon/weather-icon.component.ts [hunk 2]
```diff
@@ -10,7 +8,7 @@ import { ApiService } from '../api.service';
 })
 export class WeatherIconComponent implements OnInit {
   weatherData: any;
```
Comment (GustavBlomqvist) [inline src/app/weather-icon/weather-icon.component.ts]: Icon Value should be an `@Input`

File: src/app/weather-icon/weather-icon.component.ts
Comment (GustavBlomqvist) [inline src/app/weather-icon/weather-icon.component.ts:1]: No Need to call service in this component

**Review by GustavBlomqvist — CHANGES_REQUESTED** [review]

File: src/app/weather-icon/weather-icon.component.ts (hunk 1 above)
Comment (KlaudiaLum) [reply src/app/weather-icon/weather-icon.component.ts]: Done

File: src/app/weather-icon/weather-icon.component.ts (hunk 2 above)
Comment (KlaudiaLum) [reply src/app/weather-icon/weather-icon.component.ts]: Done

File: src/app/weather-icon/weather-icon.component.ts
Comment (KlaudiaLum) [reply src/app/weather-icon/weather-icon.component.ts:1]: Done , but now when I change value in postman the API dosen't update.

File: src/app/app.component.html
Comment (GustavBlomqvist) [inline src/app/app.component.html:1]: We should delete this empty file

File: src/app/weather-icon/weather-icon.component.html
Comment (GustavBlomqvist) [inline src/app/weather-icon/weather-icon.component.html:1]: This empty file should be deleted

File: src/app/weather-icon/weather-icon.component.ts [hunk 3]
```diff
@@ -1,42 +1,18 @@
...
 
-  constructor(private apiService: ApiService) { }
+  @Input() weatherIconValue: number = 0; 
 
- 
-  ngOnInit(): void {
-    this.generateRandomCoordinates();
-    this.getWeather();
-  }
+ constructor() { }
```
Comment (GustavBlomqvist) [inline src/app/weather-icon/weather-icon.component.ts]: We don't need this empty constructor

**Review by GustavBlomqvist — APPROVED** [review]
Looks really good. Nice cleanup. 

The value for the weather icon is only read on startup so that's why it's not updating.
This is fine for now. I've added a few comment, only just cleanups. LGTM

**Review by GustavBlomqvist — APPROVED** [review]

**Pull request merged**